java/ql/test/query-tests/security/CWE-611/SAXSourceTests.java

```diff
@@ -0,0 +1,41 @@
+import java.net.Socket;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Unmarshaller;
+
+import org.xml.sax.InputSource;
+import org.xml.sax.XMLReader;
+import org.xml.sax.helpers.XMLReaderFactory;
+
+public class SAXSourceTests {
+
+  public void unsafeSource(Socket sock) throws Exception {
+    XMLReader reader = XMLReaderFactory.createXMLReader();
+    SAXSource source = new SAXSource(reader, new InputSource(sock.getInputStream()));
+    JAXBContext jc = JAXBContext.newInstance(Object.class);
+    Unmarshaller um = jc.createUnmarshaller();
+    um.unmarshal(source); //unsafe
```

Comment:
Use GOOD and BAD for uniformity, and say briefly why these examples are safe or not?

Reply:
```suggestion
    um.unmarshal(source); // BAD
```

Reply:
I'm not really familiar enough with these code snippets to add much further information (beyond saying "because the query says so"). I just copied these tests to make them public in order to more easily adjust their expected output in other dataflow work.